scripts/genomes_cluster.py

```diff
@@ -0,0 +1,72 @@
+
+'''
+Creates full-genome fasta files for each clusters.
+'''
+import os
+import json
+import argparse
+
+'''
+Load clusters into usable python dictionary
+'''
+def clusters(file):
+    diction = {}
+    with open(file) as jfile:
+        json_data = json.load(jfile)
+        for strain, cluster_dict in json_data["nodes"].items():
+            cluster_number = cluster_dict['cluster']
+            if cluster_number in diction:
+                diction[cluster_number][strain] = []
+            else:
+                diction[cluster_number] = {strain : []}
+    return diction
+
+
+'''
+Adds sequences into python dictionary by cluster
+'''
+def sequences(clusters, files):
+    for fname in files:
+        with open(fname) as jfile:
+            json_data = json.load(jfile)
+            for cluster_id, cluster_strains in clusters.items():
+                for strain in cluster_strains.keys():
+                    if strain in json_data["nodes"].keys():
+                        seq = json_data["nodes"][strain]["sequence"]
+                        clusters[cluster_id][strain].append(seq)
+    for cluster_id, cluster_strains in clusters.items():
+        for strain, sequence in cluster_strains.items():
+            clusters[cluster_id][strain] = ''.join(sequence)
+    return clusters
+
+
+'''
+Outputs python dictionary sequences into fasta files
+'''
+def genomes(seq_dict, lineage, resolution):
+    for cluster, cluster_strains in seq_dict.items():
+        with open ('results/clusters/'+lineage+'_'+resolution+'/'+cluster+'.fasta', "w") as fasta: 
```

Comment:
Inclusion of `--lineage` and `--resolution` is certainly a workable solution. However, I think you'll have both a more flexible script and a script that is more consistent with other elements of the snakefile if you specify output in `genome_cluster.py` as `--output` and pass in `results/clusters/{lineage}_{resolution}` from the `checkpoint clusters_fasta` snakemake rule.

Reply:
I agree with the consistency with the rest of the workflow. However; this PR with the checkpoint will not actually run, at least on my machine. It creates an non-zero exit code error saying error in `rule clusters_fasta`. Miguel and I have been troubleshooting this for the past day or so but haven't yet fixed it.. My workaround to this is to define output location in the script like `genomes_{lineage}_cluster{cluster}_{resolution}.fasta` where `{cluster}` is defined by the variable cluster in the script, and then in `rule all` ask for the output for `cluster = 0`.

Reply:
I'm not sure how to use this workaround if `--output` is defined in the snakefile because,, as I understand it, we couldn't actually name the output with the correct cluster number.